certbot-apache/certbot_apache/configurator.py

```diff
@@ -884,65 +921,77 @@ def _copy_create_ssl_vhost_skeleton(self, avail_fp, ssl_fp):
 
         try:
             with open(avail_fp, "r") as orig_file:
-                with open(ssl_fp, "w") as new_file:
-                    new_file.write("<IfModule mod_ssl.c>\n")
-
-                    comment = ("# Some rewrite rules in this file were "
-                              "disabled on your HTTPS site,\n"
-                              "# because they have the potential to create "
-                              "redirection loops.\n")
+                orig_file_list = [line for line in orig_file]
+                if vhost_num != -1:
+                    orig_file_list = self._create_block_segments(orig_file_list, vhost_num)
 
-                    for line in orig_file:
-                        A = line.lstrip().startswith("RewriteCond")
-                        B = line.lstrip().startswith("RewriteRule")
-
-                        if not (A or B):
-                            new_file.write(line)
-                            continue
+            if ssl_fp in self._skeletons:
+                bit = "a"
+                self._skeletons[ssl_fp].append(avail_fp)
+            else:
+                bit = "w"
+                self._skeletons[ssl_fp] = [avail_fp]
+
+            with open(ssl_fp, bit) as new_file:
```

Comment:
Suppose a user has multiple vhosts in a file and he runs Certbot for only one of those domains. If he runs Certbot again for another domain, we'll currently blow away the configuration file created during his first run. I think this behavior kind of sucks and we can easily fix it.

I think we should either always use `a` (which should create the file if it doesn't exist) or append a number to the filename when necessary like `-le-ssl-2.conf` to avoid overwriting. In most situations, I think using `a` for the mode is fine, but it can certainly go wrong. The easy example is the user has a matching `-le-ssl.conf` file but it's invalid and is disabled. Here we'll add our own directives, enable it, Apache will fail, and we'll have to revert our changes.